naxsi_src/naxsi_runtime.c

```diff
@@ -1096,7 +1132,60 @@ ngx_http_output_forbidden_page(ngx_http_request_ctx_t *ctx,
             json[2] = '\0';
             break;
           }
-          strcat(json, value);
+          // Escape it according to the RFC. JSON is pretty liberal: The only characters you must escape are \, ", and control codes (anything less than U+0020)
+          // Control codes should not be here
+          char* result = NULL;
+          int empty = 0;
+          int escaped = 0;
+
+          if(strstr(value,"\\"))
+          { 
+            //escaping JSON
+            result = replace_str(value,"\\","\\\\");
+            if(!result)
+              empty = 1;  
+
+            if(empty || strlen(json)+strlen(result)>NGX_MAX_ERROR_STR-100-3)
```

Comment:
Please use a macro and add some parenthesis like:
`if (empty || ((strlen(json) + strlen(result)) > (NGX_MAX_ERROR_STR-100-3)) )`

Reply:
@wargio do you mean like: 

#define max_json(json,result) ((strlen(json) + strlen(result)) > (NGX_MAX_ERROR_STR-100-3)))

?

Reply:
yes, that would be lovely.

Reply:
OK